cpp/modules/deck.gl/core/src/lib/attribute/attribute-manager.h

```diff
@@ -61,7 +61,7 @@ class AttributeManager {
   auto getNeedsRedraw(bool clearRedrawFlags = false) -> bool;
   void setNeedsRedraw();
 
-  void add(const std::shared_ptr<lumagl::AttributeDescriptor>& descriptor);
+  void add(const std::shared_ptr<AttributeDescriptor>& descriptor);
```

Comment:
I  am not sure it should be possible to add and remove attributes (this would require rebuilding the model pipeline) so this could be an array in the constructor.

Reply:
Attribute descriptors should be passed by value and copied. No need to manage them with shared pointers!